### PR TITLE
APS-212: Allow Tasks to be filtered by area

### PR DESCRIPTION
**Pull request by vubalasu**
# Context

# Changes in this PR
Allow Tasks to be filtered by area
### Before
Screenshot 2024-01-26 at 16 44 39

### After
Screenshot 2024-01-26 at 16 45 37

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Have all changes to any dependencies been deployed to both `preprod` and
      `production` in a backwards compatible way? (This includes our API,
      infrastructure, third-party integrations and any secrets or environment variables)
- [ ] Has a required data migration been included in this change or been done in
      advance?

## Post merge checklist

Once we've merged we now need to release this through to production before
considering it done.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-temporary-accommodation-ui)
and work through the following steps:

- [ ] Has the product manager asked to provide approval for this feature?
  - [ ] Has the product manager approved?
- [ ] Manually approve release to preprod
- [ ] Verify change released to preprod
- [ ] Manually approve release to prod
- [ ] Verify change released to production

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

File: server/controllers/tasksController.ts [hunk 1]
```diff
@@ -10,11 +10,13 @@ export default class TasksController {
   constructor(
     private readonly taskService: TaskService,
     private readonly applicationService: ApplicationService,
+    private readonly apAreaService: ApAreaService,
   ) {}
 
   index(): TypedRequestHandler<Request, Response> {
     return async (req: Request, res: Response) => {
       const allocatedFilter = (req.query.allocatedFilter as AllocatedFilter) || 'allocated'
+      const apAreaId = req.query.areas ? req.query.areas : res.locals.user.apArea
```
Comment (pezholio) [inline server/controllers/tasksController.ts:19]: Thanks for putting this in. Can we get a test for this behaviour?

**Review by pezholio — CHANGES_REQUESTED** [review]

File: server/controllers/tasksController.test.ts [hunk 2]
```diff
@@ -22,20 +29,24 @@ describe('TasksController', () => {
...
 
   beforeEach(() => {
     jest.resetAllMocks()
-    tasksController = new TasksController(taskService, applicationService)
+    tasksController = new TasksController(taskService, applicationService, apAreaService)
   })
 
   describe('index', () => {
     it('should render the tasks template', async () => {
+      const requestWithQuery = { ...request, query: { areas: '1234' } }
```
[the remark this answers is not in the document]
Comment (vubalasu) [reply server/controllers/tasksController.test.ts:43]: ok you want move defalut one to here

[the remark this answers is not in the document]
Comment (pezholio) [reply server/controllers/tasksController.test.ts:43]: Please 👍

**Review by pezholio — APPROVED** [review]
LGTM 👍

**Pull request merged**